### PR TITLE
Refactor toast in presentation-uploader

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

It refactors the toast to not depend on the presentation-uploader component, instead, it will be a separate component that will watch any presentation coming from the service of presentation uploader, and so, can be used in every ocasion the client wants to send a document. Moreover, it will notify the presenter of possible conversions that might be occurring, since it is watching the `Presentations` collection as well.

### Closes Issue(s)

#15505

### More

**Credits** and **huge thanks** to @JoVictorNunes, who made a patch to insert the count of pages in the conversion part, closing issue #15505:

https://github.com/GuiLeme/bigbluebutton/pull/6

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-toast/presentation-toast-controller.js [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+export const ToastController = ({ intl }) => {
+
+	useTracker(() => {
+
+		const presentationsRenderedFalseAndConversionFalse = Presentations.find({ $or: [{renderedInToast: false}, {"conversion.done": false}] }).fetch();
+		const convertingPresentations = presentationsRenderedFalseAndConversionFalse.filter(p => !p.renderedInToast )
+		let tmpIdconvertingPresentations = presentationsRenderedFalseAndConversionFalse.filter(p => !p.conversion.done)
+			.map(p => p.tmpPresId)
+		// tmpIdconvertingPresentations = UploadingPresentations.find({}).fetch().filter(p => tmpIdconvertingPresentations.includes(p.tmpPresId))
+		//   .map(p => p.tmpPresId)
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-toast/presentation-toast-controller.js:276]:
```suggestion
```

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-toast/presentation-toast-controller.js [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+		UploadingPresentations.find({}).fetch().filter(p => tmpIdconvertingPresentations.includes(p.tmpPresId))
+			.map(p => UploadingPresentations.remove({tmpPresId: p.tmpPresId}));
+		// Remove all presentations from the uploading collection if they are already
+		// converting.
+		// UploadingPresentations.remove({tmpPresId: {$all: tmpIdconvertingPresentations}});
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-toast/presentation-toast-controller.js:281]:
```suggestion
```

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js [hunk 3]
```diff
@@ -73,6 +77,9 @@ const observePresentationConversion = (
   tmpPresId,
   onConversion,
 ) => new Promise((resolve) => {
+
+
+  // Fazer o onConversion modificar a upload e ver o que mais da pra fazer
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js:82]:
```suggestion
```

Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js:82]:
```suggestion
```

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx [hunk 4]
```diff
@@ -613,6 +580,7 @@ class PresentationUploader extends Component {
             logCode: 'presentationuploader_component_save_error',
             extraInfo: { error },
           }, 'Presentation uploader catch error on confirm');
+          console.log("Erro aqui....", error)
```
Comment (JoVictorNunes) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx:583]:
```suggestion
```

**Review by JoVictorNunes — APPROVED** [review]
The client part seems all ok! Let's hold on @gustavotrott's review on the backend part.

**Review by gustavotrott — CHANGES_REQUESTED** [review]
image(14)

I can still reproduce the problem related in #15259.
Although this PR brings great improvements to the pres-upload toast! If you want, we can merge this changes and try to fix the problem in a subsequent PR. It seems to be related with Tldraw component only.

**Comment by danielpetri1** [comment]
Awesome stuff, after some testing it looks like https://github.com/bigbluebutton/bigbluebutton/issues/15259 and https://github.com/bigbluebutton/bigbluebutton/issues/15773 are fixed now with the latest commits! 🎉🎉 With the progress showing too!🥳

When sending a presentation to chat, however, the toast message doesn't go away once it converts. Any clues as to why that could be?

**Pull request merged**